// models/recruitment.js
var mongoose = require('mongoose');

var Schema = mongoose.Schema;

var recruitmentSchema = new Schema({
    recruitmentNeeds: {
        deathknight: {
            class: String,
            specs:{
                blood: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                frost: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                unholy: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
        demonhunter: {
            class: String,
            specs:{
                vengeance: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                havoc: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
        druid: {
            class: String,
            specs:{
                guardian: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                restoration: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                feral: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                balance: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
        hunter: {
            class: String,
            specs:{
                beastmastery: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                marksman: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                survival: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
        mage: {
            class: String,
            specs:{
                arcane: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                frost: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                fire: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
        monk: {
            class: String,
            specs:{
                windwalker: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                mistweaver: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                brewmaster: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
        paladin: {
            class: String,
            specs:{
                holy: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                protection: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                retribution: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
        priest: {
            class: String,
            specs:{
                shadow: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                holy: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                discipline: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
        rogue: {
            class: String,
            specs:{
                subtlety: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                combat: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                assassination: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
        shaman: {
            class: String,
            specs:{
                enhancement: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                restoration: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                elemental: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
        warlock: {
            class: String,
            specs:{
                destruction: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                affliction: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                demonology: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
        warrior: {
            class: String,
            specs:{
                protection: {
                    spec: String,
                    desire: String,
                    amount: Number
                },
                fury: {

                    spec: String,
                    desire: String,
                    amount: Number
                },
                arms: {

                    spec: String,
                    desire: String,
                    amount: Number
                }
            },
            anySpec: Boolean,
            total:  Number
        },
    },
    guild: String
})

module.exports = mongoose.model('recruitment', recruitmentSchema);
